### ISSUE TITLE
Some questions about the code.

### ISSUE THREAD

**Issue by Rango-T10000** [wenchaozheng/WRF-GS#6]
Thanks for your wonderful work.
When I read your paper and code, I am confused with some questions. I really hope to get your answers.

1. In your paper, a  Mercator projection model is proposed to handle the problem that  the RX antenna receives signals from a hemispherical direction which is different from the cameras in optical 3D-GS. I think it's an excellent solutions and really smart. As a result, your projection model build a one to one mapping between a pixel in spectrum and an AoA of the signal. But I can not find that in the code, in /train.py:
```
        r_o = scene.r_o
        gateway_orientation = scene.gateway_orientation 
        R = torch.from_numpy(Rotation.from_quat(gateway_orientation).as_matrix()).float()
        tx_pos = tx_pos.cuda()
        viewpoint_cam = generate_new_cam(R, r_o)
        gaussians.dist_signal_mapping(tx_pos, r_o)




        # Render
        if (iteration - 1) == debug_from:
            pipe.debug = True


        bg = torch.rand((3), device="cuda") if opt.random_background else background


        render_pkg = render(viewpoint_cam, gaussians, pipe, bg, use_trained_exp=dataset.train_test_exp, separate_sh=SPARSE_ADAM_AVAILABLE)
        image, viewspace_point_tensor, visibility_filter, radii = render_pkg["render"], render_pkg["viewspace_points"], render_pkg["visibility_filter"], render_pkg["radii"]
```

So which part is about the Mercator projection?

2. According to the paper,  two MLPs take the 3D Gaussians position and TX position as input, and output the signals and attenuation value to initialize the Gaussian primitives.

<img width="845" height="503" alt="Image" src="https://github.com/user-attachments/assets/53d42d8a-90ac-4f4a-a361-903a5ba4fcec" />

The corresponding code is:
```
    def dist_signal_mapping(self, tx, r_o):
        r_o = torch.tensor(r_o).cuda()
        pts = self._xyz
        view = self._xyz - tx
        tx = tx.repeat(pts.size(0), 1)
        raw = self.network_fn(pts, tx, view)
        path = torch.norm(self._xyz - r_o)
        path_loss = 0.025/path
        att_a, att_p, s_a, s_p = raw[...,0], raw[...,1], raw[...,2], raw[...,3] # [batchsize]
        att_p, s_p = torch.sigmoid(att_p)*np.pi*2, torch.sigmoid(s_p)*np.pi*2
        att_a, s_a = abs(F.leaky_relu(att_a)), abs(F.leaky_relu(s_a))    
        s_p = torch.exp(1j*s_p) 
        att_p = torch.exp(1j*att_p) 
        s_a = s_a.unsqueeze(1)
        att_a = att_a.unsqueeze(1)
        att_p = att_p.unsqueeze(1)
        
        '''Different from the paper, opacity is used here to replace the attenuation component of the network output.'''
        signal = s_a*att_a/path_loss
        att = att_a*att_p
        # signal = signal*att
        signal = torch.cat((signal, signal, signal),-1)
        self._features_dc = signal.unsqueeze(-2)
        # self._opacity = att

```
In the code, the network actually has three inputs, but not two inputs like the paper claims. The third input is view = self._xyz - tx, which is the direction from tx to 3DGS, I cannot understand why does the MLP need to take this as input?

What's more, 
```
        signal = s_a*att_a/path_loss
```
But the paper claim that signals and attenuation in space are typically expressed as complex numbers, so it should be:
 signal = s_a * s_p ?, why do you use signal = s_a*att_a/path_loss?

And in the end, att is totally useless, I am very comfused. So it is to be consistent with the original optical 3DGS code and to facilitate the direct use of its render function?

My last question is that self._features_dc is nn.Parameter, which is a leaf and learnable parameter when created. But here it is directly changed as the MLP's output, it is then not a leaf and learnable parameter. Therefore, during the entire training process, this parameter representing signal is updated based on the output of the MLP, rather than being a learnable parameter like the color parameters in 3DGS.
I am very confused about that.

**Comment by ceciliaxxy** [comment]
Thank the author for sharing this excellent work! I've been studying the code and have some similar questions about the implementation:
1. Mercator Projection
I couldn't locate the Mercator projection implementation in the code. 
2. Unused Output Parameter
The att output doesn't seem to be used anywhere. What's its intended purpose?
3. Parameter Training Logic
I'm confused about self._features_dc - it's an nn.Parameter but also gets modified by MLP outputs. How does gradient computation work in this case?
4. View Direction Calculation
For the MLP input view = self._xyz - tx, shouldn't this be view = self.ro - tx to represent the direction from receiver to points?
5. Point Cloud Data Source
The code initializes points randomly instead of loading actual point clouds. I can not find the real 3D data. I am confused with the similar questions.

**Comment by wenchaozheng** [comment]
Thank you for your attention to WRF-GS. 
Firstly, regarding the Mercator projection, it is implemented in the computeCov2D and related functions under the path of submodules/diff-gaussian rasterization/CUDA_raster/forwards. cu. 
Secondly， regarding the issue of input and output of the network, the original intention of this model is to enable the network to distinguish the different signal properties of each Gaussian at different TX positions. Therefore, directly using the network's output as the result for calculation is one of the methods. (Of course, it may not be as efficient) At the same time, for the attenuation attribute, we found that the opacity attribute can completely replace the representation. 
Therefore, in WRF-GSplus, we have improved the relevant structures and achieved superior performance. We also welcome you to follow our WRF-GSplus and hope it provides you with more inspiration.
